### DISCUSSION TITLE
Error handling - How to catch all unhandled errors?

### DISCUSSION THREAD

**Discussion by Dr-Phone** [bukhalo/nestjs-telegraf#150] [Q&A]
Hi,

First of all, thanks for this great library. I am planning to use it in production and before I do, I need to figure out how to catch all unhandled errors. For this, Telegraf documentation suggest the following -

```
bot.catch((err, ctx) => {
  console.log("Ooops, encountered an error for", ctx.updateType, err)
})
bot.start((ctx) => {
  throw new Error('Example error')
})
bot.launch()
```
On these lines, I do the following in my service:
```
constructor(@InjectBot() private bot: TelegrafProvider){
       this.bot.catch((err, ctx) => {
            console.log("Encountered an error for", ctx.updateType, 'Error:', err);
            ctx.reply("Something went wrong. Please try again later");
        });
}
```
However, this doesn't seem to work. What am I doing wrong?

**Comment by evilsprut** [comment]
Hey @Dr-Phone, thx! Module already has hardcoded catch:
https://github.com/bukhalo/nestjs-telegraf/blob/a309d8640d5a46e7a32d1298c9c1548adff7f1cf/lib/telegraf.provider.ts#L24
That's not good, i know)

The only thing I can offer at the moment is to import the sources of this module to your project and make it work the way you want it to. OpenSource power 🔥

I will think about how to solve this problem in the next release. I think [Exception Filters](https://docs.nestjs.com/exception-filters) built-in functionality of will work well for that.

And, feel free for make PR.

**Comment by Dr-Phone** [comment]
Hey @bukhalo, thanks for your quick reply. Exception Filters might actually work.

For now, I've littered the code with try-catch blocks everywhere due to the deadline pressure. I guess will have to think about it over the weekend.
